### DISCUSSION TITLE
基于openwrt的alist如何绑定域名？

### DISCUSSION THREAD

**Discussion by Cxyofficial** [AlistGo/alist#4330] [General]
我在openwrt上部署了alist，请问我该怎么样在公网通过我的域名访问alist，实现公网网盘？

**Comment by jkoor** [comment]
有公网 ip 的话，用 ddns

没公网 ip 的话，内网穿透用处不大，速度太慢

**Reply by Cxyofficial** [reply]
每个人都有ipv6,我可以把ipv6ddns到域名吗？

**Reply by jkoor** [reply]
当然可以，不过访问的前提是两端都是 ipv6

**Reply by Cxyofficial** [reply]
啥意思？

**Reply by jkoor** [reply]
就是你在别处，也得用 ipv6 网络访问

**Reply by Cxyofficial** [reply]
还是有点不明白

**Comment by wang33winner** [comment]
openwrt你有没有公网IPv4?可以`curl -4 ip.sb`查看ipv4公网ip或者`curl -6 ip.sb`查看ipv6公网ip，看一下这些ip 有没有在你的ifconfig网卡上，以确认是否有公网ip
然后自己搜索学习一下ddns把自己的域名映射到对应公网ip，openwrt下有很多实现插件或脚本。
另外就是如果用ipv6，你在外网的的客户端，也要支持ipv6（现在ipv6还不一定普及，有些地方联网是没有ipv6，此时就不发通过ipv6访问）

**Reply by ctempleton3** [reply]
I am sorry that I don’t speak Chinese. So I am posting in English.

I am running alist on my openwrt router. I have done everything you have suggested for the original question. I have a domain name, ddns setup, acme.h certs for that domain. When I go into Lucy it wants an IP address not a domain name. How do I bind alist to a domain name or an interface (like my wan). I can manually plug in my wan ip into the config but when it changes in time it won’t be right. I have looked thru support documentation and can’t find how to do this.

Thank you in advance for your help.

**Reply by Cxyofficial** [reply]
You can try cloudflared tunnel?
